aten/src/ATen/native/cuda/ScatterGatherKernel.cu

```diff
@@ -421,6 +506,32 @@ void scatter_reduce_cuda_kernel(const Tensor& self, const int64_t dim, const Ten
   }
 }
 
+void scatter_reduce_two_cuda_kernel(const Tensor& self, const int64_t dim, const Tensor& index,
+                                    const Tensor& src, const SCATTER_GATHER_OP& reduce) {
+  switch (reduce) {
+  case SCATTER_GATHER_OP::REDUCE_ADD :
+    cuda_scatter_gather_base_kernel<true, false>()(self, dim, index, src,
+            "scatter_reduce_cuda_sum_", reduce_add);
+    break;
+  case SCATTER_GATHER_OP::REDUCE_MULTIPLY :
+    cuda_scatter_gather_base_kernel<true, false>()(self, dim, index, src,
+            "scatter_reduce_cuda_prod_", reduce_multiply);
+    break;
+  case SCATTER_GATHER_OP::REDUCE_MAXIMUM :
+    cuda_scatter_gather_base_kernel<true, false>()(self, dim, index, src,
+            "scatter_reduce_cuda_amax_", reduce_maximum);
+    break;
+  case SCATTER_GATHER_OP::REDUCE_MINIMUM :
+    cuda_scatter_gather_base_kernel<true, false>()(self, dim, index, src,
+            "scatter_reduce_cuda_amin_", reduce_minimum);
+    break;
+  case SCATTER_GATHER_OP::REDUCE_MEAN :
+    cuda_scatter_gather_base_kernel<true, false>()(self, dim, index, src,
```

Comment:
given that ReduceMean functor is literally the same as ReduceAdd functor, why are we instantiating kernels for both?